src/test/java/com/feedzai/commons/sql/abstraction/dml/result/ResultIteratorTest.java

```diff
@@ -0,0 +1,57 @@
+package com.feedzai.commons.sql.abstraction.dml.result;
+
+import com.feedzai.commons.sql.abstraction.engine.DatabaseEngineException;
+import mockit.Expectations;
+import mockit.Mocked;
+import org.junit.Test;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Class for testing the result iterator behaviour.
+ *
+ * @author Helena Poleri (helena.poleri@feedzai.com)
+ * @since 2.7.1
+ */
+public class ResultIteratorTest {
```

Comment:
(probably for this comment is overkill) but, you could create a AbstractResultIteratorTest and then one for each engine that would provide the correct result iterator for the test.
I ok with this since for a simple test I don't think we should have that overhead but calling out @marpereira to check that.

Reply:
discussed offline